Mention that text fragments are less durable

Josh-Cena opened the pull request:
Fix https://github.com/mdn/content/issues/39342

Josh-Cena committed:
Mention that text fragments are less durable

dipikabh commented on files/en-us/web/uri/reference/fragment/text_fragments/index.md:
````diff
@@ -28,9 +28,9 @@ The issue with linking to specific document fragments is that the author of the
 ```
 
-If the ID is changed or removed, the document fragment is ignored, and the link just links through to the top of the page. This is reasonable in terms of graceful degradation, but it would arguably be better if the author of the link had full control over where they link to, without needing to rely on the page author.
+Not all documents have such anchors, and even if they do, depending on the context, linking to a heading may be much less obvious than linking directly to the piece of text you are citing. This is where text fragments come in: it allows the link author to have full control over where in the document they want to link to, without requiring any special markup in the document itself. For example, a  ...
````
- Let's bring back the deleted para to maintain continuity with the previous text. We can follow that up with your additions

> If the ID is changed or removed, the browser will ignore the fragment and will open the top of the page. This is reasonable in terms of graceful degradation, but it also means that you need to depend on the page's markup.

- Some suggestions to remove few words to make prose more compact
```suggestion
Not all documents have such anchors, and even if they do, linking to a heading might be much less obvious than linking directly to the specific text you are citing. This is where **text fragments** help: they allow the link author to have full control over what text to link to, without requiring any special markup in the target document. For example, a search engine may refer to a specific sentence in its search results, and clicking the link will take you directly to that sentence.
```

dipikabh commented on files/en-us/web/uri/reference/fragment/text_fragments/index.md:
```diff
@@ -28,9 +28,9 @@ The issue with linking to specific document fragments is that the author of the
+Not all documents have such anchors, and even if they do, depending on the context, linking to a heading may be much less obvious than linking directly to the piece of text you are citing. This is where text fragments come in: it allows the link author to have full control over where in the document they want to link to, without requiring any special markup in the document itself. For example, a  ...
 
-**Text fragments** make this a reality — they allow link authors to specify text content to link to, rather than document fragments, in a flexible manner.
+Of course, text fragments also have the limitation that text in a document is less stable than document structure. If the linked document is updated, the fragment would not navigate to the correct point, but would stay at the top of the page. This is fine for transient links such as those in search results, but if you intend for the link to work over time, document fragments may be more reliable.
```
```suggestion
However, text fragments also have a limitation: text in a document is less stable than document structure. If the text in the linked document is updated, the fragment will no longer match, and the browser will open the top of the page. This is fine for transient links such as those in search results, but if you intend for the link to work over time, document fragments may be more reliable.
```

dipikabh commented on files/en-us/web/uri/reference/fragment/text_fragments/index.md:
while here, can we also improve the intro para - the 1st sentence is long and convoluted and the flow of sentences is breaking up by the browser sentence.

> **Text fragments** let you link directly to specific text in a web page, without requiring the page author to add an ID. They use a special syntax in the URL fragment. This feature lets you create deep links to content you don't control, without relying on the presence of IDs. It also makes sharing links more useful, since you can directly point others to specific words. Browsers may differ in how they draw attention to the linked text (for example, with a color highlight) and/or scroll it into view.

dipikabh reviewed (approved):
Thanks a lot! It's looking good overall

Josh-Cena committed:
Apply suggestions from code review

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

Josh-Cena committed:
Improve intro

Revised the explanation of text fragments to clarify functionality and benefits.

dipikabh reviewed (approved):
Just one nit, and then it's good to go. Thanks a lot!

Josh-Cena committed:
Update files/en-us/web/uri/reference/fragment/text_fragments/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>
